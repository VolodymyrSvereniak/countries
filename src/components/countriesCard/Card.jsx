import styled from "styled-components";

const Wrapper = styled.article`
  width: 15.4rem;
  background-color: var(--colors-ui-base);
  border-radius: var(--radius);
  box-shadow: var(--shadow);
  cursor: pointer;
  overflow: hidden;
`;

const CardImage = styled.img`
  display: block;
  width: 100%;
  height: 147px;
  object-fit: cover;
  box-shadow: var(--shadow);
`;

const CardBody = styled.div`
  padding: 0.9375rem;
`;

const CardTitle = styled.h3`
  padding: 0.7rem 0.3rem 0;
  margin: 0;
  font-weight: var(--fw-bold);
`;

const CardInfo = styled.ul`
  list-style: none;
  margin: 0;
  padding: 1rem 0 1rem;
  font-size: var(--fs-sm);
`;

const CardInfoItem = styled.li`
  font-weight: var(--fw-light);
  /* line-height: 1.5; */
  padding: 0.3rem;
`;

const CardInfoTitle = styled.b`
  font-weight: var(--fw-bold);
`;

export default function Card({ img, name, info = [], handleNavigate }) {
  return (
    <Wrapper onClick={handleNavigate}>
      <CardImage src={img} alt={name} />
      <CardBody>
        <CardTitle>{name}</CardTitle>
        <CardInfo>
          {info.map((el) => (
            <CardInfoItem key={el.title}>
              <CardInfoTitle>{el.title}:</CardInfoTitle> {el.description}
            </CardInfoItem>
          ))}
        </CardInfo>
      </CardBody>
    </Wrapper>
  );
}
