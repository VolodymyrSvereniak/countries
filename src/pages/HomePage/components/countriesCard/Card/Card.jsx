import {
  Wrapper,
  CardImage,
  CardBody,
  CardTitle,
  CardInfo,
  CardInfoItem,
  CardInfoTitle
} from "./CardStyled";

export default function Card({ img, name, info = [], handleNavigate }) {
  return (
    <Wrapper onClick={handleNavigate}>
      <CardImage src={img} alt={name} />
      <CardBody>
        <CardTitle>{name}</CardTitle>
        <CardInfo>
          {info.map((el) => (
            <CardInfoItem key={el.title}>
              <CardInfoTitle>{el.title}:</CardInfoTitle> {el.description}
            </CardInfoItem>
          ))}
        </CardInfo>
      </CardBody>
    </Wrapper>
  );
}
