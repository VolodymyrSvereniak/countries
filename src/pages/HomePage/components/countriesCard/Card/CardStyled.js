import styled from "styled-components";
import { LazyLoadImage } from "react-lazy-load-image-component";


export const Wrapper = styled.article`
  width: 15.4rem;
  background-color: var(--colors-ui-base);
  border-radius: var(--radius);
  box-shadow: var(--shadow);
  cursor: pointer;
  overflow: hidden;
`;

export const CardImage = styled(LazyLoadImage)`
  display: block;
  width: 100%;
  height: 9.1875rem;
  object-fit: cover;
  box-shadow: var(--shadow);
`;

export const CardBody = styled.div`
  padding: 0.9375rem;
`;

export const CardTitle = styled.h3`
  padding: 0.7rem 0.3rem 0;
  margin: 0;
  font-weight: var(--fw-bold);
`;

export const CardInfo = styled.ul`
  list-style: none;
  margin: 0;
  padding: 1rem 0 1rem;
  font-size: var(--fs-sm);
`;

export const CardInfoItem = styled.li`
  font-weight: var(--fw-light);
  /* line-height: 1.5; */
  padding: 0.3rem;
`;

export const CardInfoTitle = styled.b`
  font-weight: var(--fw-bold);
`;