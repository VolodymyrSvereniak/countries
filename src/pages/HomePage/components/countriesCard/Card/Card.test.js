import { screen, render, fireEvent } from "@testing-library/react";
import Card from "./Card";


const navigateClick = jest.fn();


const data = {
  img: "https://flagcdn.com/w320/ua.png",
  name: "Ukraine",
  info: [
    {
      title: "Population",
      description: "44,134,693",
    },
    {
      title: "Region",
      description: "Europe",
    },
    {
      title: "Capital",
      description: "Kyiv",
    },
  ],
  handleNavigate: navigateClick,
};

describe("Render cards", () => {
  test("Render card", () => {
    render(<Card {...data} />);
    
    expect(screen.getByText("Ukraine")).toBeInTheDocument();
  });
  
  test("Card snapshot", () => {
    const card = render(<Card {...data}/>)
    
    expect(card).toMatchSnapshot();
  });
  
  test("Empty card snapshot", () => {
    const card = render(<Card />)
    
    expect(card).toMatchSnapshot();
  })

  test("Button navigate to country details", () => {
    render(<Card {...data}/>)
    const article = screen.getByRole('article');

    fireEvent.click(article);


    expect(navigateClick).toHaveBeenCalled()
  })
});
