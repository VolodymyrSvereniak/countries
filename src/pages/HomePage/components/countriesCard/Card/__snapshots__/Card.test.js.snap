// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render cards Card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="sc-aYaIB YGarY"
      >
        <img
          alt="Ukraine"
          class="sc-gEvDqW bRQHKc"
          src="https://flagcdn.com/w320/ua.png"
        />
        <div
          class="sc-eqUzNf cOpjBe"
        >
          <h3
            class="sc-fqkwJk iXqJUu"
          >
            Ukraine
          </h3>
          <ul
            class="sc-dcJtft iOWBnM"
          >
            <li
              class="sc-iGgVNO knpHwG"
            >
              <b
                class="sc-gsFSjX jVlTiE"
              >
                Population
                :
              </b>
               
              44,134,693
            </li>
            <li
              class="sc-iGgVNO knpHwG"
            >
              <b
                class="sc-gsFSjX jVlTiE"
              >
                Region
                :
              </b>
               
              Europe
            </li>
            <li
              class="sc-iGgVNO knpHwG"
            >
              <b
                class="sc-gsFSjX jVlTiE"
              >
                Capital
                :
              </b>
               
              Kyiv
            </li>
          </ul>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="sc-aYaIB YGarY"
    >
      <img
        alt="Ukraine"
        class="sc-gEvDqW bRQHKc"
        src="https://flagcdn.com/w320/ua.png"
      />
      <div
        class="sc-eqUzNf cOpjBe"
      >
        <h3
          class="sc-fqkwJk iXqJUu"
        >
          Ukraine
        </h3>
        <ul
          class="sc-dcJtft iOWBnM"
        >
          <li
            class="sc-iGgVNO knpHwG"
          >
            <b
              class="sc-gsFSjX jVlTiE"
            >
              Population
              :
            </b>
             
            44,134,693
          </li>
          <li
            class="sc-iGgVNO knpHwG"
          >
            <b
              class="sc-gsFSjX jVlTiE"
            >
              Region
              :
            </b>
             
            Europe
          </li>
          <li
            class="sc-iGgVNO knpHwG"
          >
            <b
              class="sc-gsFSjX jVlTiE"
            >
              Capital
              :
            </b>
             
            Kyiv
          </li>
        </ul>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render cards Empty card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="sc-aYaIB YGarY"
      >
        <img
          class="sc-gEvDqW bRQHKc"
        />
        <div
          class="sc-eqUzNf cOpjBe"
        >
          <h3
            class="sc-fqkwJk iXqJUu"
          />
          <ul
            class="sc-dcJtft iOWBnM"
          />
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="sc-aYaIB YGarY"
    >
      <img
        class="sc-gEvDqW bRQHKc"
      />
      <div
        class="sc-eqUzNf cOpjBe"
      >
        <h3
          class="sc-fqkwJk iXqJUu"
        />
        <ul
          class="sc-dcJtft iOWBnM"
        />
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
